libsolutil/CommonIO.cpp

```diff
@@ -132,18 +132,11 @@ int solidity::util::readStandardInputChar()
 
 string solidity::util::absolutePath(string const& _path, string const& _reference)
 {
-	boost::filesystem::path p(_path);
-	// Anything that does not start with `.` is an absolute path.
-	if (p.begin() == p.end() || (*p.begin() != "." && *p.begin() != ".."))
-		return _path;
-	boost::filesystem::path result(_reference);
-	result.remove_filename();
-	for (boost::filesystem::path::iterator it = p.begin(); it != p.end(); ++it)
-		if (*it == "..")
-			result = result.parent_path();
-		else if (*it != ".")
-			result /= *it;
-	return result.generic_string();
+	// TODO: Handle throw of boost::filesystem::filesystem_error
+	return boost::filesystem::canonical(
+		boost::filesystem::path(_path),
+		boost::filesystem::path(_reference)
```

Comment:
This leaves the file name in reference. But I guess removing it brings us back to the original problem :)
```suggestion
		boost::filesystem::path(_reference).remove_filename();
```

Anyway, I see this PR more as a refactor than a replacement for #10976. I think we should do both but #10976 should go in first because it's a fix.